src/System/Console/GetOpt/Generics.hs

```diff
@@ -227,9 +222,18 @@ helpWrapper progName modifiers args fields =
     toOptDescrUnit = map (fmap (const ()))
 
     header :: String
-    header = progName ++ " " ++ "[OPTIONS]" ++
-      maybe "" (\ t -> " [" ++ map toUpper t ++ "]")
-        (getPositionalArgumentType modifiers)
+    header = ((progName ++ " ") ++) $ unwords $
```

Comment:
can't you put `progName` after `unwords` and then avoid the space and extra `++`s?